Frontend/src/Components/Battery/Battery.js

```diff
@@ -0,0 +1,143 @@
+    const chargeEnableColor = (props.data?.charge_enable[0] ?? -1.0) ? getColor("greenBg", colorMode) : getColor("redBg", colorMode);
+    const chargeEnableTextColor = (props.data?.charge_enable[0] ?? -1.0) ? getColor("txtGreenBg", colorMode) : getColor("txtRedBg", colorMode);
+
+    const fanSpeedMin = ((props.data?.pack_temp[0] ?? -1.0) < (20 + 5 * CONSTANTS.fan_speed.MAX)) ?
+                         Math.floor(((((props.data?.pack_temp[0] ?? -1.0) >= 20) ? props.data?.pack_temp[0] : 15) - 15) / 5) :
+                         CONSTANTS.fan_speed.MAX;
+
+    return (
+        <Flex flex="auto" direction="column" overflowY="scroll">
+            <RangeRow
+                dataTitle="Pack Temp"
+                dataValue={props.data?.pack_temp[0] ?? -1.0}
+                dataConstant={CONSTANTS.pack_temp}
+            />
+            <RangeRow
+                dataTitle="Pack Internal Temp"
+                dataValue={props.data?.pack_internal_temp[0] ?? -1.0}
+                dataConstant={CONSTANTS.pack_internal_temp}
+
+                bg={headerBg}
+            />
+            <Center
+                borderTopWidth={1}
+                borderBottomWidth={1}
+                borderRightWidth={1}
+                borderColor={borderColor}
+                lineHeight='1.2em'
+                height='100%'
+            >
+                <Flex w='95%'>
+                    <Center
+                        border='2px'
+                        borderRadius='0px'
+                        h='75%'
+                        background={fanEnableColor}
+                        marginTop='1%'
+                        marginBottom='1%'
+                        bg={((props.data?.fan_speed[0] ?? -1.0) < fanSpeedMin || (props.data?.fan_speed[0] ?? -1.0) > CONSTANTS.fan_speed.MAX) ?
+        getColor("errorBg", colorMode) : headerBg}
+                    >
+                        <Text fontSize='xs' margin='0.7px 2px 0.7px'><strong>{fanEnableText}</strong></Text>
+                    </Center>
+                    <Text fontSize='sm' lineHeight='2em' marginLeft='2.5%'>Fan Speed:</Text>
+                    <Spacer />
+                    <Text fontSize='sm' paddingRight='0.5em' lineHeight='2em'>
+                        {props.data?.fan_speed[0] ?? -1}
+                    </Text>
+
+                    {/* Hold the RangeBar within a vertical Flex to vertically center it */}
+                    <Flex direction='column'>
+                        <Spacer />
+                        <RangeBar 
+                            w={props.rangeBarW ?? '3em'} 
+                            h='1em'
+                            borderRadius='0px' 
+
+                            val={(props.data?.fan_speed[0] ?? -1.0)}
+                            min={fanSpeedMin}
+                            max={CONSTANTS.fan_speed.MAX}
+                        />
+                        <Spacer />
+                    </Flex>
```

Comment:
```suggestion
                    <RangeBar 
                        w='3em'
                        h='1em'
                        borderRadius='0px'

                        val={(props.data?.fan_speed[0] ?? -1.0)}
                        min={fanSpeedMin}
                        max={CONSTANTS.fan_speed.MAX}
                    />
```
- The Flex and Spacers aren't needed because the outer Flex for the row vertically centers its elements (see comment with `alignItems='center'`).
- There is no `rangeBarW` prop being passed to the component, so that can be removed.
- Very minor, but there is trailing whitespace in the `w` and `borderRadius` lines.